README.md

```diff
@@ -276,6 +293,46 @@ The rule configuration files in this package (`index.js`, `typescript.js`, etc)
 
 ESLint offers several [ways to disable a rule for a line or file](https://eslint.org/docs/user-guide/configuring/rules#disabling-rules). Suppressions should be as targeted as possible and should include a comment explaining the suppression.
 
+### Migrating to ESLint Flat Configuration Format
+
```

Comment:
Some more notes to include in this section:
1. mention the `@stylistic` rules that changed names. Clients may need to modify rule configurations in their config file or rule suppressions in their source to use the new names.
2. mention the workaround you mentioned at UI WG about needing to explicitly link to the config file from angular.json
3. mention the strictNullChecks change. Clients with strictNullChecks enabled can remove rules they had previously overridden. Clients with strictNullChecks disabled may need to add configurations to maintain previous behavior
4. In the PR change files, link to this section instead of mentioning just the strictNullChecks part of the change.